common/src/main/java/net/william278/husksync/listener/EventListener.java

```diff
@@ -0,0 +1,163 @@
+package net.william278.husksync.listener;
+
+import net.william278.husksync.HuskSync;
+import net.william278.husksync.config.Settings;
+import net.william278.husksync.data.ItemData;
+import net.william278.husksync.data.DataSaveCause;
+import net.william278.husksync.data.UserData;
+import net.william278.husksync.player.OnlineUser;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Level;
+
+public abstract class EventListener {
+
+    /**
+     * The plugin instance
+     */
+    protected final HuskSync huskSync;
+
+    /**
+     * Set of UUIDs current awaiting item synchronization. Events will be cancelled for these users
+     */
+    private final HashSet<UUID> usersAwaitingSync;
+
+    /**
+     * Whether the plugin is currently being disabled
+     */
+    private boolean disabling;
+
+    protected EventListener(@NotNull HuskSync huskSync) {
+        this.huskSync = huskSync;
+        this.usersAwaitingSync = new HashSet<>();
+        this.disabling = false;
+    }
+
+    public final void handlePlayerJoin(@NotNull OnlineUser user) {
+        if (user.isDead()) {
+            return;
+        }
+        usersAwaitingSync.add(user.uuid);
+        CompletableFuture.runAsync(() -> {
+            try {
+                // Hold reading data for the network latency threshold, to ensure the source server has set the redis key
+                Thread.sleep(Math.min(0, huskSync.getSettings().getIntegerValue(Settings.ConfigOption.SYNCHRONIZATION_NETWORK_LATENCY_MILLISECONDS)));
```

Comment:
Should this not be Math.max instead? Cause here it will always sleep for 0ms

Reply:
You're absolutely right!